src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -254,29 +253,54 @@ protected async Task<(Guid, string projectName, SourceText)> GetGuidAndProjectNa
                 // in cases like Any-Code (which does not use a VSWorkspace).  So we are tolerant
                 // when we have another type of workspace.  This means we will show results, but
                 // certain features (like filtering) may not work in that context.
-                var workspace = document.Project.Solution.Workspace as VisualStudioWorkspace;
+                var vsWorkspace = document.Project.Solution.Workspace as VisualStudioWorkspace;
 
                 var projectName = document.Project.Name;
-                var guid = workspace.GetProjectGuid(document.Project.Id);
+                var guid = vsWorkspace?.GetProjectGuid(document.Project.Id) ?? Guid.Empty;
 
                 var sourceText = await document.GetTextAsync(CancellationToken).ConfigureAwait(false);
                 return (guid, projectName, sourceText);
             }
 
-            protected async Task<Entry> CreateDocumentSpanEntryAsync(
+            protected async Task<Entry> TryCreateDocumentSpanEntryAsync(
                 RoslynDefinitionBucket definitionBucket,
                 DocumentSpan documentSpan,
                 HighlightSpanKind spanKind)
             {
                 var document = documentSpan.Document;
                 var (guid, projectName, sourceText) = await GetGuidAndProjectNameAndSourceTextAsync(document).ConfigureAwait(false);
+                var excerptResult = await ExcerptAsync(sourceText, documentSpan).ConfigureAwait(false);
 
-                var classifiedSpansAndHighlightSpan =
-                    await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(documentSpan, CancellationToken).ConfigureAwait(false);
+                var mappedDocumentSpan = await AbstractDocumentSpanEntry.TryMapAndGetFirstAsync(documentSpan, sourceText, CancellationToken).ConfigureAwait(false);
+                if (mappedDocumentSpan == null)
+                {
+                    return null;
+                }
```

Comment:
when does this happen?  Won't this mean that a reference will be hidden from teh user?

Reply:
yep. razor want to hide span that it can't map for now. they are thinking changing thier side to map everything but for now, they can't and they want to filter out ones it can't like FAR already doing for hidden code span.

anyway, this is by design.

Reply:
existing code already returned null and handled null.